ksql-engine/src/main/java/io/confluent/ksql/function/udf/string/MaskKudf.java

```diff
@@ -27,24 +28,23 @@
 
   @Udf(description = "Returns a masked version of the input string. All characters of the input"
       + " will be replaced according to the default masking rules.")
-  public String mask(final String input) {
+  @SuppressWarnings("MethodMayBeStatic") // Invoked via reflection
+  public String mask(
+      @UdfParameter("input STRING to be masked") final String input
+  ) {
     return doMask(new Masker(), input);
   }
 
-  // TODO these descriptions would be much easier and more structured if we had an annotation for
-  // each param (a bit like javadoc) as it's basically the extra params that are getting described
-  // in each variant of the UDF
   @Udf(description = "Returns a masked version of the input string. All characters of the input"
-      + " will be replaced with the specified masking characters: e.g."
-      + " mask(input, upperCaseMask, lowerCaseMask, digitMask, otherMask)."
-      + " Pass NULL for any of the mask characters to prevent masking of that character type.")
-  public String mask(final String input, final String upper, final String lower, final String digit,
-      final String other) {
-    // TODO once KSQL gains Char sql-datatype support we should change the xxMask params to int
-    // (codepoint) instead of String
-
-    // TODO really need a way for UDFs to do one-shot init() stuff instead of repeating all this
```

Comment:
Removed this comment without action as its actually incorrect.  The comment assumes `MASK` will be called with literals for the masking characters. However, it could be called with the masking characters being data-driven, i.e. coming from other columns in the data.  Hence, such 1-time i initialization won't work.

The 1-time init the comment is referring to is actually just grabbing the first character from each of the supplied strings. It's hardly a big issue.

Of course, in the future, we could support some marker that meant a param must be a literal, and hence support such an optimisation, but it's hardly high on the list right now.  Only profiling will tell if this is important.